src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -395,7 +397,7 @@ def _should_skip_saving_checkpoint(self, trainer: "pl.Trainer") -> bool:
         from pytorch_lightning.trainer.states import TrainerFn
 
         return (
-            trainer.fast_dev_run  # disable checkpointing with fast_dev_run
+            bool(trainer.fast_dev_run)  # disable checkpointing with fast_dev_run
```

Comment:
why bool?

Reply:
mypy seems to be strict in this regard. It doesn't like that we check it like this if it can also be an int